src/App.js

```diff
@@ -14,11 +14,95 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+export class App extends React.Component {
+  state = {
+    nameOfProduct: 'Selected good : ',
+    arrayOfProducts: [],
+  }
 
+  constructor() {
+    super();
+    this.clearTheSelection = this.clearTheSelection.bind(this);
+    this.clickHandlerWithCtrl = this.clickHandlerWithCtrl.bind(this);
+  }
+
+  clickHandlerWithoutCtrl = (product) => {
+    this.setState(() => ({
+      nameOfProduct: `Selected good : ${product}`,
+    }));
+  }
+
+  clickHandlerWithCtrl(product) {
+    this.setState((prevState) => {
+      let manyNamesOfProducts;
+
+      if (!prevState.arrayOfProducts.includes(product)) {
+        prevState.arrayOfProducts.push(product);
+
+        if (prevState.arrayOfProducts.length === 1) {
+          manyNamesOfProducts
+           = `Selected good : ${prevState.arrayOfProducts[0]}`;
+        } else {
+          manyNamesOfProducts
+         = `Selected goods : ${prevState.arrayOfProducts[0]}`;
+          for (let i = 1; i < prevState.arrayOfProducts.length; i += 1) {
+            manyNamesOfProducts += `, ${prevState.arrayOfProducts[i]}`;
+          }
+        }
+      } else {
+        return '';
+      }
```

Comment:
use array methods to add/delete item from the array
if items already in the array - use filter to delete it from the array
if items should be added to the array, use [...prevState.arrayOfProducts, item]